src/hash.cr

```diff
@@ -171,6 +171,36 @@ class Hash(K, V)
     indexes.map {|index| self[index] }
   end
 
+  # Returns the first key with the given *value*, else raises `KeyError`.
+  #
+  # ```
+  # hash = {"foo": "bar", "baz": "qux"}
+  # hash.key("bar")    #=> "foo"
+  # hash.key("qux")    #=> "baz"
+  # hash.key("foobar") #=> Missing hash key for value: foobar (KeyError)
+  # ```
+  def key(value)
+    each do |k, v|
+      return k if v == value
+    end
+    raise KeyError.new "Missing hash key for value: #{value}"
+  end
+
+  # Returns *true* if any key pairs with the given *value*.
+  #
+  # ```
+  # hash = {"foo": "bar", "baz": "qux"}
+  # hash.key("bar")    #=> true
+  # hash.key("qux")    #=> true
+  # hash.key("foobar") #=> false
+  # ```
+  def key?(value)
```

Comment:
That's not the contract we're aiming for `key` vs `key?`. Like `Hash#[]` vs `Hash#[]?` for example the former should raise where the latter returns `nil`.